src/sqlfluff/cli/formatters.py

```diff
@@ -55,16 +55,16 @@ def format_violation(violation, max_line_length=90):
         if violation.line_no is not None:
             line_elem = f"{violation.line_no:4d}"
         else:
-            line_elem = "   -"
+            line_elem = "   -"  # pragma: no cover
         if violation.line_pos is not None:
             pos_elem = f"{violation.line_pos:4d}"
         else:
-            pos_elem = "   -"
-    else:
+            pos_elem = "   -"  # pragma: no cover
```

Comment:
@alanmcruickshank: Do you think it's possible a violation would have a `line_pos` value of `None`? I'm wondering if there could be an `assert` here. Possibly also add [type hints to `SQLBaseError.__init__`](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/core/errors.py#L16L20).